server/src/main/java/org/elasticsearch/cluster/metadata/ShutdownShardMigrationStatus.java

```diff
@@ -36,35 +36,90 @@ public class ShutdownShardMigrationStatus implements Writeable, ChunkedToXConten
     public static final String NODE_ALLOCATION_DECISION_KEY = "node_allocation_decision";
 
     private final SingleNodeShutdownMetadata.Status status;
+    private final long startedShards;
+    private final long relocatingShards;
+    private final long initializingShards;
     private final long shardsRemaining;
     @Nullable
     private final String explanation;
     @Nullable
     private final ShardAllocationDecision allocationDecision;
 
-    public ShutdownShardMigrationStatus(SingleNodeShutdownMetadata.Status status, long shardsRemaining) {
-        this(status, shardsRemaining, null, null);
+    public ShutdownShardMigrationStatus(
+        SingleNodeShutdownMetadata.Status status,
+        long shardsRemaining,
+        @Nullable String explanation,
+        @Nullable ShardAllocationDecision allocationDecision
+    ) {
+        this(status, -1, -1, -1, shardsRemaining, explanation, null);
     }
 
-    public ShutdownShardMigrationStatus(SingleNodeShutdownMetadata.Status status, long shardsRemaining, @Nullable String explanation) {
-        this(status, shardsRemaining, explanation, null);
+    public ShutdownShardMigrationStatus(
+        SingleNodeShutdownMetadata.Status status,
+        long startedShards,
+        long relocatingShards,
+        long initializingShards
+    ) {
+        this(
+            status,
+            startedShards,
+            relocatingShards,
+            initializingShards,
+            startedShards + relocatingShards + initializingShards,
+            null,
+            null
+        );
     }
 
     public ShutdownShardMigrationStatus(
         SingleNodeShutdownMetadata.Status status,
+        long startedShards,
+        long relocatingShards,
+        long initializingShards,
+        @Nullable String explanation
+    ) {
+        this(
+            status,
+            startedShards,
+            relocatingShards,
+            initializingShards,
+            startedShards + relocatingShards + initializingShards,
+            explanation,
+            null
+        );
+    }
+
+    public ShutdownShardMigrationStatus(
```

Comment:
Can this constructor be private? The constructors above this one enforce the relationship between shardsRemaining and started/relocating/initializing, either defining shardsRemaining as the sum or setting the others to -1.

Reply:
Seems like it should work. I pushed [fa15dd9](https://github.com/elastic/elasticsearch/pull/102271/commits/fa15dd9bc3eda5b8dbe2e4a1b9aece0b532abdae).